soci/ztoc_serializer.go

```diff
@@ -0,0 +1,227 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package soci
+
+import (
+	"bytes"
+	"io"
+	"sort"
+	"time"
+
+	ztoc_flatbuffers "github.com/awslabs/soci-snapshotter/soci/fbs/ztoc"
+	flatbuffers "github.com/google/flatbuffers/go"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// ZtocSerializer incapsulates the serialization logic for Ztoc.
+// Currently, it implements serialization/deserialization to/from flatbuffers.
+type ZtocSerializer struct {
+}
```

Comment:
I left it as a struct, since it has two methods: Serialize and Deserialize. Potentially the ZtocSerializer may get some state, but it's not needed as of today.

Reply:
We've talked in the past about having a separate `ztoc` package. Wouldn't it make sense to just have `ztoc.Serialize` and `ztoc.Deserialize` functions rather than these types?

Reply:
In the world as it is today, it makes sense. I wanted to keep it as type for the case if it gets some state in the future. This way, we don't need to change the invocations throughout the code.